cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -228,33 +221,44 @@ private void deleteDiskEncryptionSetOnCloud(AzureClient azureClient, String desR
                     azureClient.deleteDiskEncryptionSet(desResourceGroupName, desName);
                     LOGGER.info("Deleted {}.", description);
                     removeKeyVaultAccessPolicyFromDiskEncryptionSetServicePrincipal(azureClient, desResourceGroupName, desName,
-                            existingDiskEncryptionSet.activeKey().keyUrl(), existingDiskEncryptionSet.identity().principalId());
+                            existingDiskEncryptionSet.activeKey().keyUrl(), existingDiskEncryptionSet.identity().principalId(),
+                            existingDiskEncryptionSet.activeKey().sourceVault().id());
                 } else {
                     LOGGER.info("No {} found to delete.", description);
                 }
                 return true;
+            } catch (IllegalArgumentException e) {
+                LOGGER.error("Removing access to key vault for disk encryption set failed.", e);
+                throw e;
```

Comment:
Please remove. The retry will not work this way and the exception will not be wrapped in a `CloudConnectorException`.